CRM/Financial/BAO/FinancialItem.php

```diff
@@ -260,16 +263,18 @@ public static function checkContactPresent($contactIds, &$error) {
    * @return array
    */
   public static function getPreviousFinancialItem($entityId) {
-    $params = [
-      'entity_id' => $entityId,
-      'entity_table' => 'civicrm_line_item',
-      'options' => ['limit' => 1, 'sort' => 'id DESC'],
-    ];
-    $salesTaxFinancialAccounts = civicrm_api3('FinancialAccount', 'get', ['is_tax' => 1]);
-    if ($salesTaxFinancialAccounts['count']) {
-      $params['financial_account_id'] = ['NOT IN' => array_keys($salesTaxFinancialAccounts['values'])];
+    $financialItemAPI = FinancialItem::get(FALSE)
+      ->addWhere('entity_id', '=', $entityId)
+      ->addWhere('entity_table', '=', 'civicrm_line_item')
+      ->addOrderBy('id', 'DESC');
+
+    $salesTaxFinancialAccounts = FinancialAccount::get(FALSE)
+      ->addWhere('is_tax', '=', 1)
```

Comment:
Looks like id is all that's needed:
```suggestion
      ->addSelect('id')
      ->addWhere('is_tax', '=', 1)
```

Reply:
Agree. Updated